src/deepsparse/pipeline.py

```diff
@@ -117,25 +120,38 @@ class PipelineImplementation(Pipeline):
         other runtime information that will be used across instances of the
         MultiModelEngine to provide optimal performance when running multiple
         models concurrently
+    :param executor: An optional ThreadPoolExecutor() object, if provided the
+        pipeline executes inference requests in a non-blocking manner and returns
+        a Future object, call Future.result() on returned object to get the result.
+        Can also accept an int number of workers, a ThreadPoolExecutor object is
+        auto-initialized with the specified integer in that case; None represents
+        synchronous execution
     """
 
     def __init__(
         self,
         model_path: str,
         engine_type: str = DEEPSPARSE_ENGINE,
-        batch_size: int = 1,
+        batch_size: Optional[int] = 1,
         num_cores: int = None,
         scheduler: Scheduler = None,
         input_shapes: List[List[int]] = None,
         alias: Optional[str] = None,
         context: Optional[Context] = None,
+        executor: Optional[Union[ThreadPoolExecutor, int]] = None,
     ):
         self._model_path_orig = model_path
         self._model_path = model_path
         self._engine_type = engine_type
         self._alias = alias
-
         self.context = context
+        self._batch_size = batch_size
+
+        self.executor, self._num_async_workers = _initialize_executor_and_workers(
```

Comment:
It might be good to either raise an error or print a warning if the user passes in executor but the schemas aren't joinable/splittable. Right now it would just silently not actually used the initialized threadpool right?